LibraryServices/Services/Library.cs

```diff
@@ -0,0 +1,300 @@
+        {
+            return _authors.Find((Author) => Author.Id == id);
+        }
+
+        /// <summary>
+        /// Update Author
+        /// </summary>
+        /// <param name="id">Author's Id</param>
+        /// <param name="author">New Author</
+        public Author UpdateAuthor(int id, Author author)
+        {
+            Author authorToUpdate = _authors.Find((Author) => Author.Id == id);
+            if (authorToUpdate != null)
+            {
+                authorToUpdate.Name = author.Name;
+                authorToUpdate.Surname = author.Surname;
+                authorToUpdate.YearOfBirthday = author.YearOfBirthday;
+            }
+            return authorToUpdate;
+        }
+
+        /// <summary>
+        /// Create new Author
+        /// </summary>
+        /// <param name="author">New Author to add</param>
+        public void CreateAuthor(Author author)
+        {
+            _authors.Add(author);
+        }
+
+        /// <summary>
+        /// Delete Author from collection
+        /// </summary>
+        /// <param name="id">Author's Id</param>
+        public void DeleteAuthor(int id)
+        {
+            Author authorToDelete = _authors.Find(Author => Author.Id == id);
+            foreach (Book book in _booksInLibrary)
+            {
+                if (book.AuthorId == id)
+                {
+                    _booksInLibrary.Remove(book);
+                }
+            }
+            _authors.Remove(authorToDelete);
+        }
+
+
+        public IEnumerable<Book> GetAuthorBooks(int author_Id)
+        {
+            foreach (var authorBook in _booksInLibrary.Where(Book => Book.AuthorId == author_Id))
+            {
+                yield return authorBook;
+            }
+        }
+        #endregion
+
+        #region GenreRegion
+
+        /// <summary>
+        /// Return All Genres
+        /// </summary>
+        public IEnumerable<Genre> GetAllGenres()
+        {
+            foreach (Genre genre in _genres)
+            {
+                yield return genre;
+            }
+        }
+
+        /// <summary>
+        /// Return Genre by Id
+        /// </summary>
+        /// <param name="id">Genre's Id</param>
+        /// <returns>Genre by Id</returns>
+        public Genre GetGenre(int id)
+        {
+            return _genres.Find((Genre) => Genre.Id == id);
+        }
+
+        /// <summary>
+        /// Create new Genre
+        /// </summary>
+        /// <param name="genre">New Genre to add</param>
+        public void CreateGenre(Genre genre)
+        {
+            _genres.Add(genre);
+        }
+
+        /// <summary>
+        /// Update Genre
+        /// </summary>
+        /// <param name="id">Genre's Id</param>
+        /// <param name="genre">New Genre's Id</param>
+        /// <returns>Updated Genre</returns>
+        public Genre UpdateGenre(int id, Genre genre)
+        {
+            Genre genreToUpdate = _genres.Find((Genre) => Genre.Id == id);
```

Comment:
Genre is a local variable inside anonymous method scope. so it should start from a lower letter. - genre